articles/templates/about-template-output-targeting/about-template-output-targeting.md

```diff
@@ -0,0 +1,52 @@
+
+## Unassigned Template Output
+
+Some designers (such as the Visual Studio designer) will show unassigned Template Outputs in red:
+
+![Unassigned Template in Output Configuration Designer](images/output-config-vs-unassigned-template.png)
+
+And running the the SF while these template are unassigned will result in errors during the SF execution about there being unassigned templates.
+
+Templates can specify where their "Template Output" should be placed by default during module installation so that things work automatically, see the [Roles and Default Locations](#roles-and-default-locations) for more details.
+
+## Roles and Default Locations
+
+Templates in the module builder can have their Role and Default Location configured. These are used by Intent Architect during module installation to determine where the template's Template Output should be placed.
+
+![Module Builder Template Role Specification](images/module-builder-template-role-specification.png)
+
+Here you can specify the name of the `Role` which the Template need to be associated with so that when a Module installs this Template, it will locate that Role and place the `Template Output` in the exact same place as the `Role` so that it will not be placed by default in the root location of the `Output Configuration Designer` and be considered an unassigned `Template Output`.
+
+The `Default Location` field accepts a directory path relative to the location of where the `Role` is where the `Template Output` will be placed.
+
+The following example depicts a Template created in the Module Builder and installed in a target Application.
+
+![Module Builder Template Settings Relative Location Example](images/module-builder-template-settings-example-relative-location.png)
+
+The `Role` is set to "Distribution" and `Default Location` is set to "Controllers".
+
+![Output Configuration Relative Location Example](images/output-config-example-relative-location.png)
+
+In the example above the `Distribution` Role can be identified by the blue badge icon and on the same tree level you will find the `Controllers` Folder.
+
+Once that Module is installed, it will create the `Template Output` named "Intent.AspNetCore.Controllers.Controller" where the `Role` "Distribution" is located but in a folder location relative to the `Role` which is "Controllers".
+
+> [!NOTE]
+> In the event that a Module was installed with an unassigned `Template Output` and the `Role` was then created after the fact, then a re-installation of the same Module will relocate the unassigned `Template Output` to the location where the Role is found.
+> [!NOTE]
+> Once a `Template Output` has been placed by the user or placed next to the Role it was assigned to, Intent Architect will not adjust the `Template Output` even if the Module it originated from changed the `Role` or `Default Location`. However, if the Module containing those Templates were to be uninstalled, then those `Template Output` items will also be removed.
```

Comment:
I think both these notes are actually referring to the same aspect of the system, so it should rather be a single note which states (in the following order) that:

- Template outputs are never moved during re-installation of modules unless they were still unassigned.
- This is so that if a user decides to change the layout of their output in their application, it's not undone by re-installing/upgrading a module.

And then we don't have to really mention the whole "role being created after the fact" part and re-installation, as it's implicit if the above two points are covered.